lua/nvim-treesitter/query_predicates.lua

```diff
@@ -128,3 +128,36 @@ query.add_directive("downcase!", function(match, _, bufnr, pred, metadata)
     metadata[key] = string.lower(text)
   end
 end)
+
+-- Trim blank lines from end of the region
+-- Arguments are the captures to trim.
+query.add_directive("trim!", function(match, _, bufnr, pred, metadata)
```

Comment:
maybe `trim-lines!` would be more explicit. We might add a `trim!` latter that works like `offset!` but trims certain characters like quotes or regexes

Reply:
Perhaps, though `trim` is (IMO) well understood to mean strip whitespace around a string.

Even though it only strips full lines at the moment, the intention is that this can be improved to work more generally on any region with preceding and trailing whitespace.

For trimming other characters, I would just stick with offset or implement are more generalized substitute directive.

Reply:
yeah, when it will trim all whitespaces `trim` would be a proper name